example/lib/main.dart

```diff
@@ -1,405 +1,69 @@
-      ),
-    );
-    bool _defaultShowCopyButton = true;
+  _MyAppState createState() => _MyAppState();
+}
 
-    DartCodeViewerThemeData dartCodeViewerThemeData = DartCodeViewerTheme.of(context);
-    dartCodeViewerThemeData = dartCodeViewerThemeData.copyWith(
-      baseStyle: baseStyle ?? dartCodeViewerThemeData.baseStyle ?? _defaultBaseStyle,
-      classStyle: classStyle ?? dartCodeViewerThemeData.classStyle ?? _defaultClassStyle,
-      commentStyle: commentStyle ?? dartCodeViewerThemeData.commentStyle ?? _defaultCommentStyle,
-      constantStyle: constantStyle ?? dartCodeViewerThemeData.constantStyle ?? _defaultConstantStyle,
-      keywordStyle: keywordStyle ?? dartCodeViewerThemeData.keywordStyle ?? _defaultKeywordStyle,
-      numberStyle: numberStyle ?? dartCodeViewerThemeData.numberStyle ?? _defaultNumberStyle,
-      punctuationStyle: punctuationStyle ?? dartCodeViewerThemeData.punctuationStyle ?? _defaultPunctuationalStyle,
-      stringStyle: stringStyle ?? dartCodeViewerThemeData.stringStyle ?? _defaultStringStyle,
-      backgroundColor: backgroundColor ?? dartCodeViewerThemeData.backgroundColor ?? Theme.of(context).colorScheme.background,
-      copyButtonText: copyButtonText ?? dartCodeViewerThemeData.copyButtonText ?? _defaultCopyButtonText,
-      showCopyButton: showCopyButton ?? dartCodeViewerThemeData.showCopyButton ?? _defaultShowCopyButton,
-      height: height ?? dartCodeViewerThemeData.height ?? MediaQuery.of(context).size.height,
-      width: width ?? dartCodeViewerThemeData.width ?? MediaQuery.of(context).size.width,
-    );
+class _MyAppState extends State<MyApp> {
+  late ThemeMode _themeMode;
+  late IconData iconToggle;
 
-    return DartCodeViewerTheme(
-      data: dartCodeViewerThemeData,
-      child: Container(
-        color: dartCodeViewerThemeData.backgroundColor,
-        padding: const EdgeInsets.symmetric(horizontal: 16),
-        height: dartCodeViewerThemeData.height,
-        width: dartCodeViewerThemeData.width,
-        child: _DartCodeViewerPage(
-          codifyString(data, dartCodeViewerThemeData),
-        ),
-      ),
-    );
+  initState() {
+    super.initState();
+    _themeMode = ThemeMode.system;
+    iconToggle = Icons.toggle_on;
   }
 
-  InlineSpan codifyString(String content, DartCodeViewerThemeData dartCodeViewerThemeData) {
-    List<TextSpan> textSpans = [];
-    final codeSpans = DartSyntaxPrehighlighter().format(content);
-    // Converting CodeSpan to TextSpan by first converting to String and then TextSpan.
-    for (final span in codeSpans) {
-      textSpans.add(stringToTextSpan(span.toString(), dartCodeViewerThemeData));
+  toggleThemeMode() {
+    if (_themeMode == ThemeMode.light) {
+      _themeMode = ThemeMode.dark;
+      iconToggle = Icons.toggle_off;
+    } else if (_themeMode == ThemeMode.dark) {
+      _themeMode = ThemeMode.light;
+      iconToggle = Icons.toggle_on;
+    } else {
+      if (Brightness.light == Theme.of(context).brightness) {
```

Comment:
Maybe `Theme.of(context).brightness == Brightness.light` ?

Reply:
Done